applicationset/generators/scm_provider.go

```diff
@@ -276,6 +292,25 @@ func (g *SCMProviderGenerator) getSecretRef(ctx context.Context, ref *argoprojio
 	return string(tokenBytes), nil
 }
 
+func (g *SCMProviderGenerator) getConfigMapData(ctx context.Context, ref *argoprojiov1alpha1.ConfigMapKeyRef, namespace string) ([]byte, error) {
```

Comment:
Duplicated code in the 2 providers. Can you move to the utils package? Maybe a `kubernetes.go` or `client.go`. `getSecretRef` should probably be moved as well.

Reply:
Done